docs/01-Headless/03-Directives.md

```diff
@@ -1,5 +1,120 @@
+	console.log('Directive has been executed on node ', element.id);
+	const clickListener = (event: Event) => {
+		console.log(text, event.target);
+	};
+	element.addEventListener('click', clickListener);
+	return {
+		update: (newText) => (text = newText),
+		destroy: () => {
+			node.removeEventListener('click', clickListener);
+			console.log('Destroy function executed ', element.id);
+		},
+	};
+};
+```
+
+This example shows a very simple directive that creates an event listener on `click` events that happen on the HTML element and then print the parameter `text` in the console, along with the clicked target.
+
+Both `update` and `destroy` functions are provided by the directive, as following:
+
+- `update`: gives a way to modify the original text to something new, when the context changes (see in the next section).
+- `destroy`: remove the previously created listener, so that you don't bloat the main thread with unused listeners.
+
+### How to use a directive
+
+Considering this HTML page, in which you have a `div` container `focus-element` that includes 2 `button`, an input text `clickText` and another `button` that removes the container.
+
+```html
+<div id="content">
+	<div id="focus-element">
+		<button id="test">button 1</button>
+		<button id="test2">button 2</button>
+	</div>
+	<hr />
+	<input id="clickText" type="text" />
+	<hr />
+	<button onclick="document.getElementById('focus-element').remove()">remove dom element</button>
+</div>
+```
+
+To use the directive in vanilla TS you need the following steps:
+
+1. Create the directive on a DOM element. In this case, our elementwill be the `focus-element` container.
+
+```typescript
+const trackElement = document.getElementById('focus-element');
+const focusElementDirective = createMyDirective(trackElement, 'focus-element click');
+```
+
+By doing it, when you click on a `button` inside the `focus-element`, you will get a log into your console,
+
+```
+focus-element click
+HTMLButtonElement{attributes: {…}, innerHTML: "button 1", nodeType: 1, tagName: "button"}
+```
+
+2. Observe context changes and call the `update` function accordingly. For example, we want to change the text to the value of the `clickText` input when it changes.
+
+```typescript
+const input = document.getElementById('clickText');
+input.addEventListener('change', (event) => {
+	instance && instance.update(event.target.value);
+});
+```
+
+3. Clean up when the `focus-element` gets remove from the DOM. This can be done for example using a `MutationObserver` on the parent DOM element.
+
+```typescript
+const cleanup = (mutations: MutationRecord[]) => {
+	mutations.forEach((mutation) => {
+		for (let entry of mutation.removedNodes.entries()) {
+			if (entry.includes(trackElement) && instance) {
+				instance.destroy();
+			}
+		}
+	});
+};
+const observer = new MutationObserver(cleanup);
+observer.observe(document.getElementById('content'), {childList: true});
+```
+
+### Angular usage
```

Comment:
I don't know if it's already possible or not, so it's probably for another PR, but we'll need to have the possibilty to switch sections depending on the selected framework, so that we'll be able to see only the framework part that is specific for the reader.

Reply:
I see two possibilities here:
1) we implement some custom renderer, but then it will work only on our demo site (like samples).
2) we separate MD either duplicating the common part or creating some utils to merge MDs (but again, it won't display correctly in github) and then show the doc only if the selected framework correspond to the one for which the documentation is written. 

I would rather create an issue only for this, we need to refine the feature.